alternatives.md

```diff
@@ -0,0 +1,113 @@
+
+> A Language Server is meant to provide the language-specific smarts and communicate with development tools over a protocol that enables inter-process communication.
+
+> The idea behind the Language Server Protocol (LSP) is to standardize the protocol for how such servers and development tools communicate. This way, a single Language Server can be re-used in multiple development tools, which in turn can support multiple languages with minimal effort.
+
+LSP defines a common protocol and RPC for queries like Go-To-Definition,
+Usages, etc. But it does not define a common representation of AST because
+the goal of the project is to enable easy access to analysis that is done
+by compilers and language SDKs.
+
+Babelfish provides a common representation for ASTs allowing
+to use its output for static analysis by other tools. Queries over UAST
+structure will still allow querying for usages, etc.
+
+## [srclib](https://srclib.org/)
+
+> srclib makes developer tools like code search and static analyzers better. It supports things like jump to definition, find usages, type inference, and documentation generation.
+
+> srclib handles: package detection, global dependency resolution, type inference, querying the graph of definitions and references in code, versioning using different VCS systems, and semantic blaming.
+
+<!-- TODO -->
+
+## [ctags](http://ctags.sourceforge.net/)
+
+> Ctags generates an index (or tag) file of language objects found in source files that allows these items to be quickly and easily located by a text editor or other utility. A tag signifies a language object for which an index entry is available (or, alternatively, the index entry created for that object).
+
+Both Babelfish and Ctags provide positional information for identifiers,
+classes, directives, etc.
+
+The main difference is that Ctags does not provide any form of AST, while
+Babelfish provides native language AST as well as language-independent UAST.
+
+Also, Babelfish ecosystem allows making more complex queries over AST.
+
+## [ANTLR](https://github.com/antlr/antlr4)
+
+> ANTLR (ANother Tool for Language Recognition) is a powerful parser generator for reading, processing, executing, or translating structured text or binary files. It's widely used to build languages, tools, and frameworks. From a grammar, ANTLR generates a parser that can build parse trees and also generates a listener interface (or visitor) that makes it easy to respond to the recognition of phrases of interest.
+
+ANTLR is a language generator with support for most existing languages
+and data formats. But parsers usually produce parse trees that are usually
+different from native language AST, thus requiring additional processing
+to be used for analysis. The structure of parse tree is also language-dependent.
+
+Babelfish provides a correct native language AST as well as
+language-independent UAST for all supported languages.
```

Comment:
May be worth mentioning here that bblfsh architecture would allow to build native drivers that use ANTLR grammars instead, but on top of difference \w native language AST, that would have the burden of supporting the grammar (or something like that?).

This would highlight (and may be link to?) nice aspects of bblfsh architecture.

Reply:
Also the generated parser/lexer are language dependant. AFAIK